training/backends/kubernetes.js

```diff
@@ -161,10 +182,35 @@ module.exports = async function execTask(job, spec) {
                                     memory: (Config.TRAINING_MEMORY_USAGE + 100) + 'Mi'
                                 }
                             },
-                            volumeMounts: {}
+                            volumeMounts: [
+                                {
+                                    name: 'config',
+                                    mountPath: '/etc/almond-cloud',
+                                    readOnly: true
+                                },
+                                {
+                                    name: 'training-config',
+                                    mountPath: '/etc/almond-cloud/training',
+                                    readOnly: true
+                                },
+                            ],
+                            securityContext: {
+                                capabilities: {
+                                    add: [ 'SYS_ADMIN', 'NET_ADMIN']
```

Comment:
By the way, this securityContext setting is wrong: we should not add capabilities uncondtionally to the job process (that's the same as running as root). Instead, we should make sure that the `bwrap` executable in the image is setuid/has file capabilities, and then set `allowPrivilegeEscalation: true` here.

Reply:
I was planning to only add the capabilities for the bwrap job.  Will try out `allowPrivilegeEscalation: true` but I think we will still need [ 'SYS_ADMIN', 'NET_ADMIN'] . Even with root,  bwrap will fail to run in the container w/o these capabilities.

Reply:
Again, that should not be the case: bwrap should acquire those capabilities on its own (using setuid / file capabilities), and then drop them right after.

Reply:
I'll give it a try. I somehow remember bwrap will not even run  before it gets to call setuid.

Reply:
It didn't work.  
deployment.yaml
```
        securityContext:
          allowPrivilegeEscalation: true
```
inside the pod
```
[almond-cloud@training-996849579-r4sdx ~]$ ls -l /usr/bin/bwrap
-rwsr-xr-x 1 root root 53872 Jul 11  2018 /usr/bin/bwrap
[almond-cloud@training-996849579-r4sdx ~]$ bwrap
bwrap: capset failed: Operation not permitted
```

Reply:
Let me know if I'm missing something.  Looking at the source, it seems `set_required_caps ` was called before `setuid`.
https://github.com/containers/bubblewrap/blob/93047df6052034989a576de55a6fe2cce05f3524/bubblewrap.c#L760

Reply:
So, after reading more about docker, it turns out I was wrong. The capabilites setting is not about ambient capabilities (capabilities that are applied to all processes in a subtree regardless of the user), it is about the capability bounding set (the maximum set of capabilities that a process can obtain in any form - through setuid and file caps). And for some reason, docker drops a bunch of caps, including sys_admin and net_admin, from the bounding set (which is normally everything even for unprivileged processes). The confusing part was that documentation usually assumes the process in the container runs as root - in which case capability bounding set and effective set are the same. But we don't run as root, so leaving the capability active in the bounding set is fine. Or in other words, you're right and we need the setting.

Reply:
Glad it makes sense now. It was quite confusing to me as well.  We only need these additional CAPs for jobs with bwrap.  Let me know if you want to add a a field `additionalCapabilities` to the job spec. Otherwise, we can keep it as is.

Reply:
Let's keep it as it is if it works.